src/renderers/shared/fiber/ReactFiberBeginWork.js

```diff
@@ -268,6 +272,7 @@ module.exports = function<T, P, I, TI, C>(
       // Abort and don't process children yet.
       return null;
     } else {
+      pushHostContext(workInProgress.type);
```

Comment:
Should we pass props too? We have cases like `props.is` where props determine the type. I could imagine this being useful.